src/ansi-c/library/cprover_contracts.c

```diff
@@ -0,0 +1,3808 @@
+/// \file cprover_contracts.c
+/// This file defines types functions supporting dynamic frames in contracts.
+
+/* FUNCTION: __CPROVER_contracts_car_create */
+
+#ifndef __CPROVER_contracts_write_set_t_defined
+#  define __CPROVER_contracts_write_set_t_defined
+
+/// \brief A conditionally writable range of bytes.
+typedef struct __CPROVER_contracts_car_t
+{
+  /// \brief  True iff __CPROVER_w_ok(lb, size) holds at creation
```

Comment:
Please move these to a header file and include that each time rather than defining it multiple times in this file.

Reply:
Moving the type definitions to `ansi-c/cprover_builtin_headers.h` only works at build time.

A runtime loading the functions that depend on these types using `cprover_library_factory` fails.

The function definitions are loaded from `cprover_library_entryt cprover_library[];`.

Each `cprover_libary_entryt` is generated from a `/* FUNCTION: ... */` section extracted from the files found in `src/ansi-c/library/*.c`. These entries need to be self-contained, i.e. include all type declarations needed to parse and typecheck the function.

```c
/* FUNCTION: foo */
typedef ... some_type_t;
int foo(some_type_t param)
{
  // ...
}
``` 

So with type declarations living in `cprover_builtin_headers.h` the entries cannot be loaded at runtime, unless there's a way to inject them automatically during the elaboration of the entries.

Reply:
A workaround is to place the whole library code under a fake function name 

```c
/* FUNCTION: __CPROVER_contracts_library */
// whole library code
EOF
```
And load that all at once.